src/ebonite/build/docker.py

```diff
@@ -89,6 +91,19 @@ def get_runner(self):
         return self.default_runner
 
 
+    # TODO not sure about signature, maybe pass provider here. also, need to separate kwargs for provider, for builder, and for builder kwargs
```

Comment:
For me provider is an implementation detail of builder (without it builder would have too many responsibilities and be difficult to test) so creating it here is perfectly fine.